dart/lib/src/browser_client.dart

```diff
@@ -41,19 +42,19 @@ class SentryBrowserClient extends SentryClient {
     );
   }
 
-  SentryBrowserClient._(SentryOptions options, {String origin, String platform})
+  SentryBrowserClient._(SentryOptions options,
+      {String origin, @required String platform})
       : super.base(
           options,
-          origin: origin,
-          sdk: Sdk(name: browserSdkName, version: sdkVersion),
-          platform: platform,
+          transport: Transport(
+            dsn: options.dsn,
+            httpClient: options.httpClient,
+            clock: options.clock,
+            compressPayload: false,
```

Comment:
```suggestion
            compressPayload: options.compressPayload,
```

Reply:
as the payload compression is not available in browser should we let the choice to the user ?

Reply:
ok, I missed that, if the browser client doesn't support that, we should not give an option for sure, but I feel this should be documented in the `compressPayload` field declaration then.

Reply:
is `compressPayload` really an options ? 🤔
Could you choose to send a non compressed payload if you're on a platform which supports it ?

Reply:
we could use a conditional import in `transport.dart` to import the bodyEncoder strategy ? and let it be a transport "hidden" implementation details ?

Reply:
I forgot that it's already a conditional import in `transport.dart`. the question is : should we hide `compressPayload ` this from the public API and set it based on the current platform ?

Reply:
not really, for flutter, they still can opt in-out of the compressPayload, lets not make this limitation.
lets keep uncompressed for web, and later we fix this too https://github.com/getsentry/sentry-dart/projects/1#card-47870976

Reply:
it's not an option in Java, is it ? https://github.com/getsentry/sentry-java/blob/67669e1c6d727af618424b7542e35db5c7797c54/sentry/src/main/java/io/sentry/transport/HttpTransport.java#L254-L256 
As Flutter web needs the same behavior than other Dart web projects, I don't really understand why the user could set it ?

Reply:
its not, but it should, we have this in our Java backlog

Reply:
Could just be a no-op for Web (Dart or Flutter), since it doesn't apply.
Conditionally having the option will make it harder for users to consume the options since they'll need to conditionally set a value depending on the platform, feels like lots of friction

Reply:
a condition for importing dart or web http packages already exist, so the only difference here is if they use `web`, and set  `compressPayload=true` won't have any effect till we implement it

Reply:
I'm aware, but there's no gzip on web, the browser will gzip or not and we don't control it

Reply:
if this is not controlled on the client, totally in for NoOp, I thought this was a limitation because we didn't implement it

Reply:
we will address this on other PR https://github.com/getsentry/sentry-dart/projects/1#card-47942558